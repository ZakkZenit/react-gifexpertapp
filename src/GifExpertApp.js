import React, { useState } from 'react'
import { AddCategory } from './componentes/AddCategory';
import { GifGrid } from './componentes/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr/>

            <ol>
                {
                    categories.map(c =>
                        <GifGrid
                            key={c} 
                            category={c}
                         />
                    )
                }
            </ol>
        </>
    )
}
