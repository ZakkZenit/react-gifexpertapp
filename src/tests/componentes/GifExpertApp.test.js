import React from 'react';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../../GifExpertApp';
import {renderHook} from '@testing-library/react-hooks';
import { useFetchGifs } from '../../hooks/useFetchGifs';

describe('Pruebas en GifExpertapp', () => {
    
    test('debe cargar correctamente', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe retornar el estado inicial ', async() => {
        const {result, waitForNextUpdate} = renderHook( () => useFetchGifs('Dragon Ball'));
        const {data, loading} = result.current;

        await waitForNextUpdate();
        
        expect(data).toEqual([]);
        expect(loading).toBeTruthy();
    })
    
    test('debe retornar arreglo de imagenes y loading en false', async() => {
        
        const {result, waitForNextUpdate} = renderHook( () => useFetchGifs('Dragon Ball'));
        await waitForNextUpdate();
        
        const {data, loading} = result.current;

        expect(data.length).toBe(10);
        expect(loading).toBeFalsy();
    })
    
    
})
