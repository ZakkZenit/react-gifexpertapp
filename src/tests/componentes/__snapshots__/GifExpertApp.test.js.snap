// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertapp debe cargar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
